notebook/tests/selenium/test_shutdown.py

```diff
@@ -0,0 +1,13 @@
+"""Tests shutdown of the Kernel."""
+
+def test_shutdown(notebook):
+    notebook.edit_cell(content="print(21)")
+    notebook.browser.find_element_by_xpath('//a[text()="Kernel"]').click()
+    notebook.browser.find_element_by_id('shutdown_kernel').click()
+    notebook.browser.find_element_by_class_name('btn.btn-default.btn-sm.btn-danger').click()
+
+    notebook.wait_for_element_availability("cell")
+    notebook.execute_cell(0)
+
+    assert not notebook.is_kernel_running()
+    assert notebook.get_cell_output() == None
```

Comment:
Does this test actually fail if you comment out the lines to shut the kernel down? I.e. does the output normally appear quickly enough that it would be there before Selenium looks for it?

Reply:
@takluyver I commented the kernel shutdown statements and its corresponding assertion, and the test failed because the output was generated.

I didn't add an implicit or explicit wait since it could lead more to a race condition. And I could use something like `wait_for_element_availability` but that method depends of the output element to actually appear, which is something we don't expect since the kernel is shutdown.

Reply:
Yup, no problem that it's not properly waiting - I don't know of any good way to test that something *doesn't* happen asynchronously. So long as it's reasonably likely to fail when the relevant lines are commented out, it's still a useful test.